### ISSUE TITLE
`command-line` #:once-each doesn't error with rest args

### ISSUE THREAD

**Issue by florence** [racket/racket#1342]
How to reproduce:

Given the program "once-each.rkt":

```racket
#lang racket
(displayln 
 (command-line
  #:once-each
  [("-p") "thing" (displayln "-p")]
  #:args a a))
```

Run the program with the command line arguments "-p a -p"

Expected behavior:
Error about -p being provided more than once. (this happens with the arguments "-p -p")

Actual behavior:

The program outputs:
```
-p
(a -p)
```

This problem was noted in florence/cover/issues/122

**Comment by wilbowma** [comment]
So it's not clear to me that your expected behavior is in fact expected, since the final "-p" isn't considered a flag, but is an arg.

Nevertheless, I created a patch that fixes this.

**Comment by wilbowma** [comment]
See also the discussion in #1364

**Comment by wilbowma** [comment]
So all tests pass, but I can still imagine situation where this is not the desired behavior. Robby gave a good example on the pull request. What if instead we added something like `#:args-checked-flags`, or `#:args-no-flags` that gets this new behavior?

**Comment by rfindler** [comment]
I like that approach.

**Comment by rfindler** [comment]
Also: it may make sense to go poke around at the various `raco XYZ` things we have and change some of them to one of those new keywords.

**Comment by wilbowma** [comment]
Actually, I think a better interface might be to restrict individual flags, e.g., `#:once-each/args` would prevent the flag from showing up more than once even in the args list; that might introduce a combinatorial problem, as you might want the "no args" modifier for every kind of flag.

**Comment by jackfirth** [comment]
Doesn't the built in `--` flag cover the case where you want to pass a flag as an argument rather than a flag? `command -- foo -b bar -p` would interpret "foo", "-b", "bar", and "-p" as the list of arguments.

**Comment by rfindler** [comment]
Yes. But there may be scripts out there that don't already have the -- in
them.

**Comment by spdegabrielle** [comment]
is this still an issue?  Please consider adding the label `good first issue` so it is findable: 

[Issues labeled `good first issue` in in Racket GitHub repositories](https://github.com/search?q=type:issue+org:racket+state:open+label:%22good+first+issue%22)

**Comment by sorawee** [comment]
Correct me if I'm wrong, but I don't think this is a "good first issue" since the conclusion is not reached yet in the above discussion.

**Comment by spdegabrielle** [comment]
@sorawee my apologies - removed

**Comment by mflatt** [comment]
I've let this one sit every time I looked at it, because the idea of changing command-line parsing worries me. the command-line syntax is a language, after all, and changing a language tends to break something somewhere. Also, the idea that the first non-flag argument ends all flag parsing is an especially established convention that pre-dates Racket (see `getopt`), and people who understand that rule currently get what they expect.

Changing command-line parsing in `racket` (as opposed to just `racket/cmdline`) like #1364 this would likely break some scripts. The command `racket mod.rkt arg ...` makes `arg ...` arguments to `mod.rkt`, no matter how many flag-looking things are in `arg` or whether there are repetitions. Although `racket -- mod.rkt arg ...` would mean the same thing, I wouldn't write the`--`. (The rules make something like `#! /path/to/racket` work as expected on Unix.)

Of course, `racket` parsing doesn't necessarily have to be the same as `racket/cmdline`, and there could be designs different than #1364 that still achieve the original goal while breaking less things. Also, command-line parsing is one place where DWIM-ish ("do what I mean") irregular rules sometimes work well, as long as there is a way (like `--`) to get into a more regular subset.

But I also think that `racket/cmdline` has stood the test of time better than many parts of Racket, and I'm not convinced of the benefits here compared to the risks of making a change.

Is anyone particularly convinced that we should make a change here?

**Comment by wilbowma** [comment]
I don't recall why I originally ran into this, but I do remember wanting the error to be raised. I could see and argument for adding a new, opt-in version of `#:once-each` (as I proposed earlier in the thread) that causes the "desired" behaviour but retains backwards compatibility.
